Yosemite/YosemiteTests/Stores/OrderStoreTests.swift

```diff
@@ -563,6 +563,143 @@ final class OrderStoreTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    func test_optimistic_update_order_customer_note_correctly() {
+        // Given
+        let orderStore = OrderStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+        let expectedCustomerNote = ""
+
+        let originalOrder = sampleOrder().copy(customerNote: "Updated!")
+        let updatedOrder = originalOrder.copy(customerNote: expectedCustomerNote)
+
+        orderStore.upsertStoredOrder(readOnlyOrder: originalOrder, in: viewStorage)
+
+        /// Update: Expected Customer note is actually coming from `order.json` (Customer note == "")
+        network.simulateResponse(requestUrlSuffix: "orders/963", filename: "order")
+
+        // When
+        let result: Result<Networking.Order, Error> = waitFor { promise in
+            let action = OrderAction.updateOrderOptimistically(siteID: self.sampleSiteID, order: updatedOrder, fields: [.customerNote]) { result in
+                promise(result)
+            }
+            orderStore.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        let storageOrder = storageManager.viewStorage.loadOrder(siteID: sampleSiteID, orderID: sampleOrderID)
+        XCTAssertEqual(storageOrder?.customerNote, expectedCustomerNote)
```

Comment:
Nit: we can simply compare the note with the one from the updated order, so that'll be one less variable needed:

 ```suggestion
        XCTAssertEqual(storageOrder?.customerNote, updatedOrder.customerNote)
```

Reply:
Much better 🙂